client/utils/auth.py

```diff
@@ -196,15 +196,22 @@ def authenticate(cmd_args, endpoint='', force=False):
 
     return access_token
 
-def notebook_authenticate(cmd_args, force=False):
+def notebook_authenticate(cmd_args, force=False, silent=True):
     """ Similiar to authenticate but prints student emails after
-    all calls and uses a different way to get codes.
+    all calls and uses a different way to get codes. If SILENT is True,
+    it will supprese the error messege and redirect to FORCE=True
     """
     server = server_url(cmd_args)
     network.check_ssl()
     access_token = None
     if not force:
-        access_token = refresh_local_token(server)
+        try:
+            access_token = refresh_local_token(server)
+        except OAuthException as e:
+            # Account for Invalid Grant Error During make_token_post
+            if not silent:
+                raise e
+            return notebook_authenticate(cmd_args, force=True)
```

Comment:
Shouldn't this call with `silent=false`? If not, we'll basically do an infinite loop of trying to login them in. If it's failing more than twice, we should do: 

* Either call it with `silent=False` on line 214 (effectively lets it do one retry) 
* Pass in a counter instead a boolean. If the counter is >= 3 we raise the exception

Sounds like it usually just requires one retry -> but I'd be fine with any of the above